ui/app/rif/pages/domainsDetailPage/domainDetailActive/domainDetailActiveConfig/index.js

```diff
@@ -45,33 +40,77 @@ class DomainsDetailConfigurationScreen extends Component {
   }
 
+  componentDidMount () {
+    this.props.getConfiguration()
+      .then(configuration => {
+        const resolvers = Object.assign([], GET_RESOLVERS(configuration));
+        this.setState({
+          resolvers,
+          configuration,
+        });
+      });
+    this.loadResolver();
+  }
+
   componentDidUpdate (prevProps, prevState) {
-    if (prevProps.disableSelect !== this.props.disableSelect) {
-      this.setState({disableSelect: this.props.disableSelect});
+    if (prevProps.resolver.address !== this.props.resolver.address) {
+      this.timeoutToLoadResolver();
     }
   }
 
+  timeoutToLoadResolver = () => setTimeout(async () => {
+    let resolver = await this.props.getResolver(this.props.domainName, this.props.subdomainName);
+    if (resolver.pending) {
+      this.timeouts.push(this.timeoutToLoadResolver());
+    } else {
+      this.setState({
+        resolver: resolver,
+        disableSelect: resolver.pending,
+      });
+    }
+  }, WAIT_FOR_CONFIRMATION_DEFAULT);
+
+  componentWillUnmount () {
+    this.timeouts.forEach(timeout => clearTimeout(timeout));
+  }
+
+  loadResolver () {
+    this.props.getResolver(this.props.domainName, this.props.subdomainName)
+      .then(resolver => {
+        if (resolver.pending) {
+          this.timeouts.push(this.timeoutToLoadResolver());
+        } else {
+          this.setState({
+            resolver: resolver,
+            disableSelect: resolver.pending,
+          });
+        }
+      });
+  }
+
   async onChangeComboResolvers (e) {
     for (const resolverItem of e.target.children) {
       if (resolverItem.value === e.target.value) {
         const address = resolverItem.getAttribute('data-address');
-        const transactionListenerId = await this.props.setNewResolver(this.props.domainName, address);
+        const transactionListenerId = await this.props.setNewResolver(this.props.domainName, address, this.props.subdomainName);
         this.props.showTransactionConfirmPage({
           afterApproval: {
             action: () => {
               this.props.showDomainConfigPage({
                 ...this.props,
-                disableSelect: true,
-                selectedResolverAddress: address,
               });
               this.props.showToast('Waiting Confirmation');
               this.props.waitForListener(transactionListenerId)
                 .then(transactionReceipt => {
-                  this.props.showDomainConfigPage({
-                    ...this.props,
-                    disableSelect: false,
-                    selectedResolverAddress: address,
-                  });
+                  setTimeout(() => {
+                    this.props.showDomainConfigPage({
+                      ...this.props,
+                      resolver: {
+                        pending: false,
+                        address: address,
+                      },
+                    });
+                  }, 2000);
```

Comment:
This should be the constant right? the default waiting?

Reply:
Nice catch here!